addons/mail/static/src/models/link_preview.js

```diff
@@ -0,0 +1,94 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'LinkPreview',
+    recordMethods: {
+        async remove() {
+            await this.messaging.rpc({
+                route: '/mail/link_preview/delete',
+                params: { link_preview_id: this.id },
+            }, { shadow: true });
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsCard() {
+            return !this.isVideo && !this.isImage;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsDeletable() {
+            if (!this.message) {
+                return false;
+            }
+            return (
+                (this.message.author && this.message.author === this.messaging.currentPartner) ||
+                (this.message.guestAuthor && this.message.guestAuthor === this.messaging.currentGuest)
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsImage() {
+            return Boolean(this.image_mimetype || this.og_mimetype === 'image/gif');
+        },
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeIsVideo() {
+            if (!this.og_type) {
+                return clear();
+            }
+            const type = this.og_type.split('.');
+            return type.includes('video') && !this.isImage;
```

Comment:
Why split & includes? Why not just check if the type starts with `video.`? Can `video` be the second segment (or even later)? Can this actually be trusted?

And why not use the mimetype there? Or use the opengraph type for images?